docs/changes/cosmology/14988.api.rst

```diff
@@ -0,0 +1 @@
+Don't automatically register abstract classes with YAML for serialization.
```

Comment:
Since this was not out anywhere yet, this does not need an entry, but can we instead have the real changelog entry that was still missing?

Reply:
I can edit https://github.com/astropy/astropy/pull/14987 when that's merged.

Reply:
Can we merge #14987 into this PR? I don't see the point of so many PRs conflicting with each other over one unreleased feature.

Reply:
Sure.